sparse.c

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2025, Maksim Paimushkin <maxim.paymushkin.development@gmail.com>
+ * All rights reserved.
+ *
```

Comment:
Keep the Copyright line (and All rights reserved), the remainder of this notice is "abbreviated" through the SPDX line above and can be dropped.